src/DispatchesContextualization.php

```
@@ -0,0 +1,46 @@
+    // contextualize Drupal is simply a no-op here.
+    if ($context instanceof Drupal) {
+      if ($this instanceof DrupalContextualizerInterface) {
+        $this->contextualizeDrupal($context);
+      }
+
+      return;
+    }
+
+    // A custom context falls back to a contextualize<ContextName>() method
+    // resolved from its short name, when the ring defines one.
+    $method = 'contextualize' . (new \ReflectionClass($context))->getShortName();
```

PHPMD (2.15.0)

[error] Missing class import via use statement (line '38', column '38'). (undefined)

(MissingImport)